docs/installation.rst

```diff
@@ -22,42 +24,133 @@ Or from GitHub (for development builds)
     git clone https://github.com/PlasmaControl/DESC.git
     cd DESC
 
-    # OPTION 1: install with pip
+    # OPTION 1: install with pip after first creating a conda environment
+    conda create
     # standard build
     pip install -r requirements.txt
     # developer build (if you want to run tests)
     pip install -r devtools/dev-requirements.txt
 
     # OPTION 2: install with conda
-    # standard build
+    # only need to do ONE of these conda env create commands, not both
+    ## standard build
     conda env create --file requirements_conda.yml
-    # developer build (if you want to run tests)
+    ## developer build (if you want to run tests)
     conda env create --file devtools/dev-requirements_conda.yml
+
+
     conda activate desc-env
+    python setup.py install
```

Comment:
In this comment, I use "DESC" to refer to the directory of that name and `desc` to refer to the actual `desc` python module or package.

From the slack it looks like the motivation to add the `python setup.py install` command is to avoid `desc module not found error`.
To avoid this error, my suggestion is to either
- Add the path to the DESC directory to your computer's Python path. The instructions for doing so may depend on operating system.
- Or, when you want to use `desc`, open your terminals from the DESC directory. This is equivalent to changing your terminal session to DESC with some `cd path/to/DESC`. I've always used this option. Note that Jupyter Notebooks will need the following preamble before using the `desc` module. For example,
```python
# do this
import sys
import os
sys.path.insert(0, os.path.abspath("."))
relative_path_to_DESC_string = "../../"
sys.path.append(os.path.abspath(relative_path_to_DESC_string))
# before this
import desc.io
```

I do not recommend using `python setup.py install` as a means to add `desc` to the Python path.  It may confuse developers.

The command adds a frozen copy of the DESC directory you cloned from GitHub to your python path. Any changes you make as you develop code will be ignored.
For example, suppose you run `python setup.py install`, then later add `raise Exception("Prevent users from making Grids")` in the constructors of every grid in `grid.py`. Then suppose you start a terminal somewhere other than DESC path. You would expect the following code to raise that expection.
```python
conda activate desc-env
python
from desc.grid import *
lg = LinearGrid(L=2, M=2, N=2)
```
But it will not (I tested), because the `desc` module that is being used was frozen to modifications after the `python setup.py install` command.

Moreover, if you start a terminal from the DESC directory, then the code
```python
conda activate desc-env
python
from desc.grid import *
lg = LinearGrid(L=2, M=2, N=2)
``` 
will raise that expection.
This is because Python searches the local directory with highest priority for modules. In this case, the `desc` module that is being used are the live files in the DESC directory.

Reply:
For the reasons discussed above, if you have used the command `python setup.py install`, I recommend uninstalling that. The only way to do this is by removing the environment it was installed in entirely. Then to install DESC, the instructions in this PR are fine. To make explicit:
```sh
conda activate base
# remove the environment where `python setup.py install` was used
conda remove --name desc-env --all
# install desc as usual
git clone https://github.com/PlasmaControl/DESC.git
cd DESC
conda env create --file devtools/dev-requirements_conda.yml
conda activate desc-env
# if you want, add DESC directory to python path so that you can use `desc` module from any directory
export PYTHONPATH="${PYTHONPATH}:${HOME}/Documents/project/DESC/" # replace with your path to DESC directory
source ~/.bashrc
```

Reply:
Ok, thanks Kaya! I did see the problem that you mentioned. When I added a debug statement to a DESC core script, ran python3 setup.py install, and ran the code. It didn't stop at the debug breakpoint. 

So if I install DESC using the procedure that you have just explained, make changes to the source, say putting a debugger in perturbations.py, and then run my python3 script with DESC will the code stop at the debug breakpoint.

Reply:
Yes. It will certainly stop if you replace your debugger with an `Exception`. If it somehow does not stop at the debugger, my guess is it has something to do with jax and just in time (jit) compilation techniques in the code. Although in my experience, if jax & jit are the cause of some unexpected behavior, they will complain with a warning.

Reply:
Ok, I'll think about it. Thanks!